.github/workflows/test-ci.yaml

```diff
@@ -0,0 +1,228 @@
+        sudo apt update
+        sudo apt install podman
+    - name: Build scorecard image for test
+      run: |
+        CUSTOM_SCORECARD_IMG=ghcr.io/${{ github.repository_owner }}/cryostat-operator-scorecard:${{ env.TAG }} \
+        PLATFORMS=linux/amd64 \
+        MANIFEST_PUSH=false \
+        make scorecard-build
+    - name: Push scorecard image to ghcr.io for test
+      id: push-scorecard-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator-scorecard
+        tags: ${{ env.TAG }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build operator image for test
+      run: |
+        OPERATOR_IMG=ghcr.io/${{ github.repository_owner }}/cryostat-operator:${{ env.TAG }} \
+        SKIP_TESTS=true \
+        make oci-build
+    - name: Push operator image to ghcr.io for test
+      id: push-operator-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator
+        tags: ${{ env.TAG }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build bundle image for test
+      run: |
+        yq -i '.spec.template.spec.imagePullSecrets = [{"name": "registry-key"}]' config/manager/manager.yaml
+        OPERATOR_IMG=${{ steps.push-operator-to-ghcr.outputs.registry-path }} \
+        BUNDLE_IMG=ghcr.io/${{ github.repository_owner }}/cryostat-operator-bundle:${{ env.TAG }} \
+        make bundle bundle-build
+    - name: Push bundle image to ghcr.io for test
+      id: push-bundle-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator-bundle
+        tags: ${{ env.TAG }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Set up Kind cluster
+      run: |
+        kind create cluster --config=".github/kind-config.yaml" -n ci-${{ github.run_id }}
+        # Enabling Ingress
+        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+        kubectl rollout status -w deployment/ingress-nginx-controller -n ingress-nginx --timeout 5m
+    - name: Install Operator Lifecycle Manager
+      run: curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.24.0/install.sh | bash -s v0.24.0
+    - name: Install Cert Manager
+      run: make cert_manager
+    - uses: redhat-actions/podman-login@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
We should set the default token to "restrictive" (read-only) by defaults and explicity request write permissions for jobs.

See https://github.com/cryostatio/cryostat/pull/1635

Reply:
I thought this was what I did with setting the packages permission to write?

Reply:
I think it's a two-parter: set the workflow to request `packages: write` permissions, and then also remove the `GITHUB_TOKEN` secret-as-password reference so that the job uses the default token that only has the `packages: write`, rather than the PAT in the `GITHUB_TOKEN` secret that probably still has broader permissions.

Reply:
Yehh, the restrictive mode can be set for repo or org scope in settings, not workflow tho. So, this is good. 

But I think we should revert it back to the old PAT? Ali is working on the replacement here: https://github.com/cryostatio/cryostat-operator/pull/628

Just to keep the history clear :))

Reply:
ah I see ok I'll keep the old PAT :+1:

Reply:
I can't resolve comments ahha but this should be already resolved :))